modules/compute/vm-instance/main.tf

```diff
@@ -45,6 +45,18 @@ locals {
     "ENABLE"  = "TRUE"
   }
   enable_oslogin = var.enable_oslogin == "INHERIT" ? {} : { enable-oslogin = lookup(local.oslogin_api_values, var.enable_oslogin, "") }
+
+  machine_vals            = split("-", var.machine_type)
+  machine_family          = local.machine_vals[0]
+  machine_not_shared_core = length(local.machine_vals) > 2
+  machine_vcpus           = try(parseint(local.machine_vals[2], 10), 1)
+
+  smt_capable_family = !contains(["t2d"], local.machine_family)
+  smt_capable_vcpu   = local.machine_vcpus >= 2
+
+  smt_capable = local.smt_capable_family && local.smt_capable_vcpu && local.machine_not_shared_core
+  smt_enable  = var.threads_per_core != null && (var.threads_per_core == 0 && local.smt_capable || try(var.threads_per_core >= 1, false))
```

Comment:
nit: I think `set_threads_per_core` or `should_set_threads_per_core` would be a better variable name here. `smt_enable` is confusing to me because when we are setting `threads_per_core=1` (aka smt disabled) this value will be true.

Reply:
Changed. Thanks!